### PR TITLE
Implement scroll support to new connections wizard

### PR DESCRIPTION
**Pull request by javierluraschi**
Before,

screen shot 2017-01-17 at 9 59 15 am

after with scroll support:

screen shot 2017-01-17 at 9 59 06 am

### CODE REVIEW

**Comment by jjallaire** [comment]
I'd use even less vertical space for the connection types.

**Comment by javierluraschi** [comment]
Other dialogs, like projects, use this much spacing:

screen shot 2017-01-17 at 10 29 55 am

Which felt too close for the connections dialog, that said, this commit reduces the spacing as follows, which seems better to me:

screen shot 2017-01-17 at 10 28 53 am

Do we want to reduce this even more?

**Comment by jjallaire** [comment]
I think that your most recent treatment looks good.

I agree that the New Project dialog is too densely packed (@kevinushey and I have already discussed this). Let's actually introduce a constant that ensures that these values are synced.

File: src/gwt/src/org/rstudio/studio/client/workbench/views/connections/ui/NewConnectionNavigationPage.java [hunk 1]
```diff
@@ -53,4 +81,105 @@ public NewConnectionNavigationPage(String title,
...
+
+         for (int i = 0, n = pages.size(); i < n; i++)
+         {
+            final WizardPage<NewConnectionContext, ConnectionOptions> page = pages.get(i);
+            SelectorItem item = new SelectorItem(page, new ClickHandler()
+            {
+               @Override
+               public void onClick(ClickEvent event)
+               {
+                  onSelected_.execute(page);
```
Comment (jmcphers) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/connections/ui/NewConnectionNavigationPage.java]: does this need a null guard?

Comment (javierluraschi) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/connections/ui/NewConnectionNavigationPage.java]: Not really, `setSelectionHandler` is called during `addAndInitializePage` while constructing the page before any click event would reach this. Also, the eisting code does not check for this, something bad might be going on is this is actually NULL.

**Comment by javierluraschi** [comment]
Much better, sharing spacing and styles now, new dialogs:

screen shot 2017-01-17 at 11 17 35 am

screen shot 2017-01-17 at 11 17 44 am

**Comment by kevinushey** [comment]
TBH I'm still a fan of the more compact dialog. The icons and text now look too small for their container (in my eyes)

**Comment by javierluraschi** [comment]
We could **bold** both of them?

**Comment by javierluraschi** [comment]
Bolded? (Haven't committed a change yet)

screen shot 2017-01-17 at 11 32 21 am

screen shot 2017-01-17 at 11 32 30 am

**Comment by javierluraschi** [comment]
I think it looks worse bolded...

**Comment by kevinushey** [comment]
I think bold is a bit heavy -- bumping the font size up a bit might help.

It's also possible nudging the icons to the right a bit would help (I just think the vertical spacing + spacing to left of each icon should be roughly equal, or at least feel comfortable).

**Comment by javierluraschi** [comment]
It does look a bit better with the icons pushed right, to me...

screen shot 2017-01-17 at 11 39 31 am

screen shot 2017-01-17 at 11 39 37 am

**Comment by javierluraschi** [comment]
Reducing 4 more pixels... original:

screen shot 2017-01-17 at 11 49 38 am

Latest:

screen shot 2017-01-17 at 11 48 30 am

screen shot 2017-01-17 at 11 48 34 am

All good? It does look a tiny bit smaller that I would want, but is worth to fit all the project options without scrolling, at least for now.

**Comment by jjallaire** [comment]
Yes, I like the last iteration!

**Comment by kevinushey** [comment]
Looks good to me as well! (nit: I would nudge the icons back to the left ~2 pixels)

**Comment by javierluraschi** [comment]
Done.

screen shot 2017-01-17 at 11 48 30 am

screen shot 2017-01-17 at 12 50 07 pm

File: src/gwt/src/org/rstudio/studio/client/workbench/views/connections/ui/NewConnectionNavigationPage.java (hunk 1 above)
Comment (jmcphers) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/connections/ui/NewConnectionNavigationPage.java]: Great, thanks!

**Comment by jmcphers** [comment]
Thanks for the UX feedback everyone!

**Pull request merged**